codalab/worker/worker.py

```diff
@@ -147,37 +144,32 @@ def load_state(self):
                 resources=RunResources.from_dict(run_state.resources),
             )
 
+    @property
+    def should_stop_because_idle(self):
+        now = time.time()
+        if len(self.runs) > 0 or self.last_time_ran is None:
+            self.last_time_ran = now
```

Comment:
Not great that the property mutates...

Reply:
Right. I'll rename + make it not a property and add a docstring.